docs/design/crd-validation/README.md

```diff
@@ -0,0 +1,109 @@
+## Background
+
+This proposal is to provide CRD level validation for OCS Operator project. Currently, when the user (manually) changes CRD yaml file, the cluster will fail at the creation time if there is an error in the user input. It would have been nice to have a layer of validation when user submit the changes.
```

Comment:
Validation is always good, but what sorts of validations do you think we could reasonably provide here? If the user changes a CRD, they're already clearly doing something we don't anticipate. Assuming that the semantics of a change they make is incorrect seems difficult, especially before we receive their actual CR (which is your intention if I read this doc correctly).

I definitely see that modifying CRDs is error-prone and that's unfortunate. Still, this document spends a lot of time on the "I want" clause of the "As a... I want... so that" user story. Who wants this feature? What concrete benefit (with examples) can we give them?

Reply:
@egafford , 
Many of the initial bugs, according to which we started the validation project, has become invalid... One such use case we see is depicted in the following bug, can you please take a look,

[bug#1777073](https://bugzilla.redhat.com/show_bug.cgi?id=1777073)

Reply:
All right; I see the case and I appreciate the added detail on the benefit. It seems as though there is ongoing discussion / disagreement about why this feature is needed, but you've addressed the request for changes.